jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandlerCollection.java

```diff
@@ -155,9 +155,12 @@ public void mapContexts()
     @Override
     public void setHandlers(Handler[] handlers)
     {
-        super.setHandlers(handlers);
-        if (isStarted())
-            mapContexts();
+        synchronized (this)
+        {
+            super.setHandlers(handlers);
+            if (isStarted())
+                mapContexts();
+        }
```

Comment:
This is not right.
`mapContexts()` is not protected enough in other places so it's not worth to wrap it here in a synchronized block.
`super.setHandlers()` should be thread safe in the super class, and `isStarted()` is safe already.
We need instead to make `mapContexts()` thread safe, and remove the synchronized block here.

Reply:
removed sync for now. We need another approach.